### PR TITLE
Introduce lecture note fields

### PR DESCRIPTION
**Pull request by Parcly-Taxel**
Work on v1.2 begins here.

### CODE REVIEW

**Review by shuiyao-sg — COMMENTED** [review]
Do not merge the PR first until we discussed the architecture on Wed afternoon

File: src/main/java/seedu/address/model/person/Person.java [hunk 1]
```diff
@@ -2,81 +2,50 @@
...
-import java.util.Set;
-
-import seedu.address.model.tag.Tag;
 
 /**
- * Represents a Person in the address book.
- * Guarantees: details are present and not null, field values are validated, immutable.
+ * Represents an NUStudy lecture note. Its title and content are guaranteed non-null.
  */
 public class Person {
```
Comment (shuiyao-sg) [inline src/main/java/seedu/address/model/person/Person.java]: Should you change the class name to Note and refractor the rest of the code?

File: src/main/java/seedu/address/model/tag/Tag.java [hunk 2]
```diff
@@ -1,54 +0,0 @@
-package seedu.address.model.tag;
-
-import static java.util.Objects.requireNonNull;
-import static seedu.address.commons.util.AppUtil.checkArgument;
-
-/**
- * Represents a Tag in the address book.
- * Guarantees: immutable; name is valid as declared in {@link #isValidTagName(String)}
- */
-public class Tag {
```
Comment (shuiyao-sg) [inline src/main/java/seedu/address/model/tag/Tag.java:10]: Can we keep the Tag class and morph it to Subject?

File: src/main/java/seedu/address/logic/commands/AddCommand.java [hunk 3]
```diff
@@ -1,40 +1,29 @@
...
 
 import seedu.address.logic.commands.exceptions.CommandException;
 import seedu.address.model.Model;
 import seedu.address.model.person.Person;
 
 /**
- * Adds a person to the address book.
+ * Adds a lecture note to NUStudy.
  */
 public class AddCommand extends Command {
```
Comment (shuiyao-sg) [inline src/main/java/seedu/address/logic/commands/AddCommand.java]: Currently all the command classes only support the addition/deletion/edit...etc on lecture notes (Note class), but we will need to support these operations on 1. Questions 2. Revision Tasks which have similar structures but different fields and methods. With regard to this, do you think it is better if we have a AddCommand interface and three classes(AddNoteCommand, AddQuestionCommand, AddTaskCommand) to implement the interface. The same goes for other command classes?

File: src/main/java/seedu/address/ui/PersonCard.java [hunk 4]
```diff
@@ -13,45 +10,25 @@
  * An UI component that displays information of a {@code Person}.
  */
 public class PersonCard extends UiPart<Region> {
```
Comment (shuiyao-sg) [inline src/main/java/seedu/address/ui/PersonCard.java]: Might need to change the class name and refractor the rest of the code?

File: src/main/java/seedu/address/storage/JsonAdaptedPerson.java [hunk 5]
```diff
@@ -1,62 +1,37 @@
...
+import seedu.address.model.person.Content;
 import seedu.address.model.person.Person;
-import seedu.address.model.person.Phone;
-import seedu.address.model.tag.Tag;
+import seedu.address.model.person.Title;
 
 /**
  * Jackson-friendly version of {@link Person}.
  */
 class JsonAdaptedPerson {
```
Comment (shuiyao-sg) [inline src/main/java/seedu/address/storage/JsonAdaptedPerson.java]: Might need to change to class name?

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 6]
```diff
@@ -36,89 +28,32 @@ public static Index parseIndex(String oneBasedIndex) throws ParseException {
...
+     * @throws ParseException if the given {@code content} is invalid.
      */
-    public static Set<Tag> parseTags(Collection<String> tags) throws ParseException {
-        requireNonNull(tags);
-        final Set<Tag> tagSet = new HashSet<>();
-        for (String tagName : tags) {
-            tagSet.add(parseTag(tagName));
+    public static Content parseContent(String content) throws ParseException {
+        requireNonNull(content);
+        String trimmedAddress = content.trim();
```
Comment (irene-lly) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:53]: Might want to change this to trimmedContent.

**Comment by Parcly-Taxel** [comment]
_AND SO WE BEGIN!_

**Pull request merged**